src/lib/net/address.c

```diff
@@ -859,13 +842,9 @@ tor_addr_is_valid_ipv4n(uint32_t v4n_addr, int for_listening)
 int
 tor_port_is_valid(uint16_t port, int for_listening)
 {
-  /* Any port value is valid with for_listening. */
-  if (for_listening) {
-    return 1;
-  }
-
-  /* Otherwise, zero ports are invalid. */
-  return port != 0;
+/* If for_listening is 1 any port is valid.
+ * Otherwise, zero ports are invalid. */
+  return for_listening || port != 0;
```

Comment:
Is this really simpler than the original function?

I find it harder to read multi-part expressions on a single line, than the original code.